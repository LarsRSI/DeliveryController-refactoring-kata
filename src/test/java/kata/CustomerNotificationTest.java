package kata;

import kata.testdoubles.RecordingEmailGateway;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

import java.util.List;

import static kata.TestFactory.createDeliveryWithEmail;
import static kata.TestFactory.createDeliveryWithEmailForLocation;
import static kata.TestFactory.deliveryEventForDelivery;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
public class CustomerNotificationTest {

    private RecordingEmailGateway emailGateway = new RecordingEmailGateway();
    private DeliveryService deliveryService = new DeliveryService(emailGateway);

    @Test
    void asks_for_recommendation_after_delivery() {
        var delivery = createDeliveryWithEmail("test@example.com");
        var deliveryEvent = deliveryEventForDelivery(delivery, 17, 45);

        deliveryService.on(deliveryEvent, List.of(delivery));

        Approvals.verify(emailGateway.invocations());
    }

    @Test
    void informs_next_recipient_about_estimated_time_of_arrival() {
        var delivery = createDeliveryWithEmail("test@example.com");
        var deliveryEvent = deliveryEventForDelivery(delivery, 17, 45);
        var nextDelivery = createDeliveryWithEmailForLocation("next@example.com", 58.376125f,24.496687f);

        deliveryService.on(deliveryEvent, List.of(delivery, nextDelivery));

        Approvals.verify(emailGateway.invocations());
    }
}
