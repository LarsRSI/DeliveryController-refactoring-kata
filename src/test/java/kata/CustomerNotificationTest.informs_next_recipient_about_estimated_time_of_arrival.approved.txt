Arguments[to=test@example.com, subject=Your feedback is important to us, message=Regarding your delivery today at 2022-04-07 17:45. How likely would you be to recommend this delivery service to a friend? Click <a href='url'>here</a>]
Arguments[to=next@example.com, subject=Your delivery will arrive soon, message=Your delivery to [58.376125,24.496687] is next, estimated time of arrival is in 156 minutes. Be ready!]